telematic_system/telematic_rsu_management_service/src/main/java/com/telematic/telematic_rsu_management_service/data_ingestion/service/DataIngestionService.java

```diff
@@ -0,0 +1,155 @@
+
+import com.telematic.telematic_rsu_management_service.data_ingestion.handler.DataIngestionHandler;
+import com.telematic.telematic_rsu_management_service.messaging.nats.NatsMessagingClient;
+import com.telematic.telematic_rsu_management_service.model.RSUConfigStatus;
+import com.telematic.telematic_rsu_management_service.model.TRUConfigStatus;
+import com.telematic.telematic_rsu_management_service.repository.influx.InfluxDBRepository;
+import com.telematic.telematic_rsu_management_service.repository.mysql.TRUConfigStatusRepository;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Component
+@Slf4j
+public class DataIngestionService {
+    private NatsMessagingClient natsMessagingClient;
+    private final ApplicationContext applicationContext;
+    private final TRUConfigStatusRepository truConfigStatusRepository;
+    private final InfluxDBRepository influxDBRepository;
+    
+    @Value("${data.ingestion.workers-per-unit-rsu-pair:10}")
+    private int workersPerUnitRsuPair;
+
+    @Value("${data.ingestion.subject-prefix:unit.*.stream.rsu.*}")
+    private String dataIngestionSubjectPrefix;
+
+    @Value("${rsu_influx.token:}")
+    private String adminToken;
+
+    @Value("${rsu_influx.database:}")
+    private String databaseName;
+
+    private final Set<String> activePrefixes = new HashSet<>();
+    private final Map<String, List<DataIngestionHandler>> prefixHandlers = new HashMap<>();
+
+    public DataIngestionService(NatsMessagingClient natsMessagingClient, ApplicationContext applicationContext, TRUConfigStatusRepository truConfigStatusRepository, InfluxDBRepository influxDBRepository) {
+        this.natsMessagingClient = natsMessagingClient;
+        this.applicationContext = applicationContext;
+        this.truConfigStatusRepository = truConfigStatusRepository;
+        this.influxDBRepository = influxDBRepository;
+    }
+
+    public void initializeDataIngestionService() {
+        try {
+            if (adminToken != null && !adminToken.isBlank() && databaseName != null && !databaseName.isBlank()) {
+                influxDBRepository.createDatabaseIfNotExists(databaseName, adminToken);
+            } else {
+                throw new RuntimeException(String.format(
+                        "InfluxDB database creation failure: missing token or bucket name (token present? %b, bucket: '%s')",
+                        adminToken != null && !adminToken.isBlank(), databaseName));
+            }            
+        } catch (RuntimeException e) {
+            throw new RuntimeException(String.format("InfluxDB database creation failed: %s.", e.getMessage()), e);
+        }
+    }
+    
+     public void enableDataInjestionSubscriptions() {
+        try {
+            // Add unique instance ID to queue group to avoid message distribution across instances
+            String instanceId = java.util.UUID.randomUUID().toString().substring(0, 8);
+            
+            for (String newPrefixSubject : getLatestSubjectPrefixes().stream()
```

Comment:
Can we add some comments here to explain what's happening so its easier to read?
For example - Loop through and update subject names to expected formatting  and add unique instanceID identifier